grakel_replace/torch_wl_kernel.py

```diff
@@ -169,6 +169,10 @@ def __init__(self, n_iter: int = 5, *, normalize: bool = True) -> None:
 
     def _wl_iteration(self, adj: Tensor, labels: Tensor) -> Tensor:
         """Perform one iteration of the WL algorithm to update node labels."""
+        # Ensure the adjacency matrix is in COO format before coalescing
+        if adj.layout == torch.sparse_csr:
+            adj = adj.to_sparse_coo()
```

Comment:
Also interesting that you change format again here, would it not be better to stick to one layout throughout?